tests/serialization/test_dag_serialization.py

```diff
@@ -1233,6 +1233,67 @@ def test_deps_sorted(self):
             'airflow.ti_deps.deps.trigger_rule_dep.TriggerRuleDep',
         ]
 
+    def test_derived_dag_deps(self):
+        """
+        Tests DAG dependency detection, including derived classes
+        """
+        from airflow.operators.dummy import DummyOperator
+        from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+        from airflow.sensors.external_task import ExternalTaskSensor
+
+        class DerivedSensor(ExternalTaskSensor):
+            template_fields = ['external_dag_id', 'run_id']
+
+            def __init__(self, *, external_dag_id: str, run_id: str = None, **kwargs):
+                super().__init__(
+                    external_dag_id=external_dag_id, execution_delta=None, execution_date_fn=None, **kwargs
+                )
+                self.run_id = run_id
+
+        execution_date = datetime(2020, 1, 1)
+        for Sensor in [ExternalTaskSensor, DerivedSensor]:
+            with DAG(dag_id="test_derived_dag_deps_sensor", start_date=execution_date) as dag:
+                task1 = Sensor(
+                    task_id="task1",
+                    external_dag_id="external_dag_id",
+                    mode="reschedule",
+                )
+                task2 = DummyOperator(task_id="task2")
+                task1 >> task2
+
+            dag = SerializedDAG.to_dict(dag)
+            assert dag['dag']['dag_dependencies'] == [
+                {
+                    'source': 'external_dag_id',
+                    'target': 'test_derived_dag_deps_sensor',
+                    'dependency_type': 'sensor',
+                    'dependency_id': 'task1',
+                }
+            ]
+
+        class DerivedTrigger(TriggerDagRunOperator):
```

Comment:
please call this `DerivedOperator` instead of `DerivedTrigger` 

`Trigger` is a _thing_ in airflow so best not to use it here (could be confusing, and produce false positives in search)

```suggestion
        class DerivedOperator(TriggerDagRunOperator):
```